orte/mca/rml/ofi/rml_ofi_component.c

```diff
@@ -0,0 +1,1091 @@
+                    struct sockaddr_in* ep_sockaddr = (struct sockaddr_in*)orte_rml_ofi.ofi_conduits[cur_conduit].ep_name;
+                    opal_output_verbose(1,orte_rml_base_framework.framework_output,
+                            "%s port = 0x%x, InternetAddr = 0x%x  ",
+                            ORTE_NAME_PRINT(ORTE_PROC_MY_NAME),ep_sockaddr->sin_port,ep_sockaddr->sin_addr.s_addr);
+                    /*[end debug]*/
+                    OPAL_MODEX_SEND_STRING( ret, OPAL_PMIX_GLOBAL,
+                                            OPAL_RML_OFI_FI_SOCKADDR_IN,
+                                            orte_rml_ofi.ofi_conduits[cur_conduit].ep_name,
+                                            orte_rml_ofi.ofi_conduits[cur_conduit].epnamelen);
+                    if (ORTE_SUCCESS != ret) {
+                        opal_output_verbose(1, orte_rml_base_framework.framework_output,
+                                    "%s:%d: OPAL_MODEX_SEND failed: %s\n",
+                                    __FILE__, __LINE__, fi_strerror(-ret));
+                        free_conduit_resources(cur_conduit);
+                        /*abort this current transport, but check if next transport can be opened*/
+                        continue;
+                    }
+                    break;
+                case  FI_ADDR_PSMX :
+                    opal_output_verbose(1,orte_rml_base_framework.framework_output,
+                            "%s:%d In FI_ADDR_PSMX.  ",__FILE__,__LINE__);
+                    /*   Address is of type Intel proprietery PSMX */
+                     OPAL_MODEX_SEND_STRING( ret, OPAL_PMIX_GLOBAL,
+                                           OPAL_RML_OFI_FI_ADDR_PSMX,orte_rml_ofi.ofi_conduits[cur_conduit].ep_name,
+                                                    orte_rml_ofi.ofi_conduits[cur_conduit].epnamelen);
+                     opal_output_verbose(1,orte_rml_base_framework.framework_output,
+                            "%s:%d Opal modex send completed for FI_ADDR_PSMX.  ",__FILE__,__LINE__);
+                    if (ORTE_SUCCESS != ret) {
+                        opal_output_verbose(1, orte_rml_base_framework.framework_output,
+                                "%s:%d: OPAL_MODEX_SEND failed: %s\n",
+                                 __FILE__, __LINE__, fi_strerror(-ret));
+                        free_conduit_resources(cur_conduit);
+                        /*abort this current transport, but check if next transport can be opened*/
+                        continue;
+                    }
+                    break;
+                default:
+                     opal_output_verbose(1,orte_rml_base_framework.framework_output,
+                     "%s:%d ERROR: Cannot register address, Unhandled addr_format - %d, ep_name - %s  ",
+                      __FILE__,__LINE__,orte_rml_ofi.ofi_conduits[cur_conduit].fabric_info->addr_format,
+                      orte_rml_ofi.ofi_conduits[cur_conduit].ep_name);
```

Comment:
releasing all the conduit related allocations in rml_ofi_component_close() as they are created in rml_ofi_component_open()